core/db_classes/EE_Event.class.php

```diff
@@ -1427,20 +1427,20 @@ public function registrationFormUuid():? string
     public function registrationForm()
     {
         $FSC_UUID = $this->registrationFormUuid();
-        return !empty($FSC_UUID)
-            ? $this->get_many_related(
-                'Form_Section',
-                [
-                    [
-                        'OR' => [
-                            'Form_Section.FSC_UUID'      => $FSC_UUID, // top level form
-                            'Form_Section.FSC_belongsTo' => $FSC_UUID, // child form sections
-                        ]
-                    ],
-                    'order_by' => ['FSC_order' => 'ASC'],
+
+        if (empty($FSC_UUID)) {
+            return [];
+        }
+        
+        return EEM_Form_Section::instance()->get_all([
+            [
+                'OR' => [
+                    'FSC_UUID'      => $FSC_UUID, // top level form
+                    'FSC_belongsTo' => $FSC_UUID, // child form sections
                 ]
-            )
-            : [];
+                ],
+            'order_by' => ['FSC_order' => 'ASC'],
+        ]);
```

Comment:
we'll roll with this fix for now, but I don't like having other models as dependencies within another model object as we have had issues with that before.

I'll create an issue for investigating the underlying problem and hopefully we can find a better solution.

Reply:
I know there could be a better way that I didn't know 😆

Reply:
impressive that you were able to find the underlying issues within `EEM_Base`'s 6500+ LoC